scripts/configurator/parser/cmake_var.py

```diff
@@ -0,0 +1,39 @@
+def generate_cmake_var(type :str, var):
+   "factory function that associates the correct type of variable"
+   string_to_variable_class = {
+      "BOOL" : cmake_var_bool,
+      "INTERNAL" : cmake_var,
+      "FILEPATH": cmake_var,
+      "STRING" : cmake_var,
+   }
+   variable_instance = string_to_variable_class[type]
+   return variable_instance(var)
+
+
+class cmake_var:
+   "generic type of cmake_var, used when we don't need particular conversions"
+   def __init__(self, var: str) -> None:
```

Comment:
`var` should be `val`. The variable has a _value_ so we're now dealing with it.

Reply:
Ok, but since when we get the  val  from `cmake_row` it can effectively be either the value (in case of comments) or the `cmake_var`  holding it, i think it's just better to insert a setter to get the value no matter what.

Reply:
No, wait. I've been talking about the `var` parameter and `var` member in `cmake_var`. Not the "_var" suffix in the same. The variable (aka cmake_var) has a value. You can call it also `value` in order everything to be even clearer.

Reply:
Ok, makes sense. What about the getter idea? do you like it?  I think makes sense